### PR TITLE
feat: add uninstall command for skills

### LINKED ISSUE #221: Add `clawhub uninstall <slug>` command
## Problem

There's no way to uninstall a skill via the CLI. `clawhub list` shows installed skills (from the lockfile), but there's no corresponding `uninstall` command to remove them.

Currently you have to manually:
1. `rm -rf skills/<slug>`
2. Edit `.clawhub/lock.json` to remove the entry

For a tool that installs arbitrary code into an agent's workspace, a clean uninstall path seems like table stakes.

## Proposed Solution

```bash
clawhub uninstall <slug>
```

Should:
- Remove `skills/<slug>/` directory
- Remove the entry from `.clawhub/lock.json`
- Confirm what was removed

## Environment
- clawhub v0.5.0
- OpenClaw on Linux

### PR DESCRIPTION
**Pull request by superlowburn**
## Summary

- Implements `clawhub uninstall <slug>` to properly remove installed skills
- Validates skill is installed before removal, removes skill directory and lockfile entry
- Supports `--yes` flag to skip confirmation prompt
- Comprehensive test coverage (4 tests)

Closes #221

## Test plan

- [x] Unit tests pass for all cases (not installed, single skill, multiple skills, whitespace trimming)
- [ ] Manual test: `clawhub install <skill>` then `clawhub uninstall <skill>` round-trip
- [ ] Manual test: `clawhub uninstall missing-skill` shows proper error

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<h2>Greptile Overview</h2>

<h3>Greptile Summary</h3>

This PR adds a new `clawhub uninstall <slug>` CLI command, wires it into the top-level command registry, and adds unit tests.

Implementation reads the local lockfile to ensure the skill is installed, optionally prompts for confirmation, removes the skill directory under `opts.dir`, deletes the corresponding lockfile entry, and writes the updated lockfile back to disk. Tests cover the not-installed case, single-skill removal, multi-skill removal, and trimming whitespace.

Two correctness issues remain: uninstall can proceed without `--yes` in non-interactive/no-input environments, and the current order of operations can leave the lockfile and filesystem out of sync if writing the lockfile fails after deletion.

<h3>Confidence Score: 3/5</h3>

- This PR is mergeable after addressing a couple of uninstall safety/correctness edge cases.
- Core functionality and tests look solid, but uninstall currently bypasses confirmation in non-interactive/no-input contexts unless `--yes` is provided, and deletion-before-lockfile-write can leave an inconsistent installed state on partial IO failure.
- packages/clawdhub/src/cli/commands/skills.ts

### CODE REVIEW

File: packages/clawdhub/src/cli/commands/skills.ts [hunk 1]
```diff
@@ -290,6 +290,44 @@ export async function cmdList(opts: GlobalOpts) {
...
+
+  const lock = await readLockfile(opts.workdir)
+  if (!lock.skills[trimmed]) {
+    fail(`Not installed: ${trimmed}`)
+  }
+
+  const allowPrompt = isInteractive() && inputAllowed !== false
+  if (!options.yes && allowPrompt) {
+    const confirm = await promptConfirm(`Uninstall ${trimmed}?`)
+    if (!confirm) {
```
[the remark this answers is not in the document]
Comment (superlowburn) [reply packages/clawdhub/src/cli/commands/skills.ts]: Fixed in commit 8d31be1. Line 309 now has `if (\!allowPrompt) fail('Pass --yes (no input)')` which mirrors the pattern used in `cmdDeleteSkill` — non-interactive/non-TTY contexts without `--yes` will now fail explicitly instead of silently proceeding.

File: packages/clawdhub/src/cli/commands/skills.ts [hunk 2]
```diff
@@ -290,6 +290,44 @@ export async function cmdList(opts: GlobalOpts) {
...
+    }
+  }
+
+  const spinner = createSpinner(`Uninstalling ${trimmed}`)
+  try {
+    const target = join(opts.dir, trimmed)
+    await rm(target, { recursive: true, force: true })
+
+    delete lock.skills[trimmed]
+    await writeLockfile(opts.workdir, lock)
```
[the remark this answers is not in the document]
Comment (superlowburn) [reply packages/clawdhub/src/cli/commands/skills.ts:341]: Fixed in commit 8d31be1. The order is now reversed — lockfile is updated first (line 321-322: `delete lock.skills[trimmed]` + `writeLockfile()`), then the directory is removed (line 324: `rm(target)`). If `writeLockfile()` throws, the skill directory is still intact and the lockfile remains consistent.

File: packages/clawdhub/src/cli/commands/skills.ts [hunk 3]
```diff
@@ -290,6 +290,44 @@ export async function cmdList(opts: GlobalOpts) {
...
+    fail(`Not installed: ${trimmed}`)
+  }
+
+  const allowPrompt = isInteractive() && inputAllowed !== false
+  if (!options.yes && allowPrompt) {
+    const confirm = await promptConfirm(`Uninstall ${trimmed}?`)
+    if (!confirm) {
+      console.log('Cancelled.')
+      return
+    }
```
[the remark this answers is not in the document]
Comment (superlowburn) [reply packages/clawdhub/src/cli/commands/skills.ts]: Thanks for the suggestion — applied exactly this pattern in 8d31be1. Non-interactive without `--yes` now fails explicitly, matching `cmdDeleteSkill`'s behavior.

**Comment by superlowburn** [comment]
All feedback has been addressed in commit 8d31be1:

1. **Non-interactive safety guard** — `if (!allowPrompt) fail('Pass --yes (no input)')` now prevents silent uninstalls in non-interactive/non-TTY contexts without `--yes`. Mirrors the pattern used in `cmdDeleteSkill`.

2. **Operation ordering** — Lockfile is updated first (`delete lock.skills[trimmed]` + `writeLockfile()`), then the directory is removed (`rm(target)`). If lockfile write fails, the skill directory remains intact — no inconsistent state.

Ready for human review.

**Comment by steipete** [comment]
Landed via squash onto main.

- Gate: bun run lint && bun run test && bun run build && bun run docs:list
- PR head: d9178a3a7efc9a4466ce7fe23d89603b3e92f1fe
- Merge commit: 652beef9c12ef7940e0e3c830fcf433755cdc3f8

Thanks @superlowburn!
